cowprotocol/explorer
# Limit orders fees

# Summary

#Fixes: https://github.com/cowprotocol/cowswap/issues/1348

This PR shows the actual fee also for Limit Orders.

The main changes are to add the new field `executedSurplusFee` from the API and apply it for the different calculations. 


Before:
![image](https://user-images.githubusercontent.com/2352112/205070372-58cad62d-1577-47f3-865f-4592dd20e24d.png)

After:
![image](https://user-images.githubusercontent.com/2352112/205070456-ccc5a331-69cd-4b28-92d3-e473de8eda85.png)


## Not included
Note that there's a bit of repetition, for how the order info is derived for getting the different sections of the explorer, I didn't aim to do a refactor of that in this PR

I didn't do also storybook/tests in this PR

# To Test

1.  Load a Limit Order in DEV: https://protocol-explorer.dev.gnosisdev.com/goerli/orders/0x1473e457f1519494e72cdba6fda965dfdd7ea07320dad844a5461b608158c35b79063d9173c09887d536924e2f6eadbabac099f56388a9c1
2. Open the same one in this PR: look for `0x1473e457f1519494e72cdba6fda965dfdd7ea07320dad844a5461b608158c35b79063d9173c09887d536924e2f6eadbabac099f56388a9c1` 
3. Verify all fields
4. Make sure SWAPS are not broken :) 

# Background

Limit Orders have fee 0, the fee is payed from the surplus. The explorer is transparent showing the value of this fee.
